mypy/subtypes.py

```diff
@@ -1752,6 +1758,7 @@ def are_args_compatible(
     ignore_pos_arg_names: bool,
     allow_partial_overlap: bool,
     is_compat: Callable[[Type, Type], bool],
+    allow_imprecise_kinds: bool = False,
```

Comment:
I'd make this parameter (and possibly the three parameters before it, as well) keyword-only. The callsite at 1654 is pretty hard to read with all arguments passed positionally, IMHO